### ISSUE TITLE
Getting an exception when navigating away of a view that uses the ngx-joystick

### ISSUE THREAD

**Issue by AyhamJaradat** [wweitzel/ngx-joystick#25]
Hi! 👋 
      
Firstly, thanks for your work on this project! 🙂

Today I used [patch-package](https://github.com/ds300/patch-package) to patch `ngx-joystick@4.0.2` for the project I'm working on.

I used the ngx-joystick inside a chart component, whenever I navigate to another view, I got the following exception:

```
ERROR TypeError: Cannot read properties of undefined (reading 'destroy')
    at NgxJoystickComponent.ngOnDestroy (ngx-joystick.mjs:43:22)
```
<img width="488" alt="ngx-joystick issue" src="https://github.com/wweitzel/ngx-joystick/assets/8011881/349a25b7-c5c3-46b9-94ec-cb1a0e89d70b">

Here is the diff that solved my problem:

```diff
diff --git a/node_modules/ngx-joystick/esm2022/lib/ngx-joystick.component.mjs b/node_modules/ngx-joystick/esm2022/lib/ngx-joystick.component.mjs
index c514ae1..9035313 100644
--- a/node_modules/ngx-joystick/esm2022/lib/ngx-joystick.component.mjs
+++ b/node_modules/ngx-joystick/esm2022/lib/ngx-joystick.component.mjs
@@ -39,7 +39,9 @@ export class NgxJoystickComponent {
     }
     ngOnDestroy() {
         window.clearInterval(this.interval);
-        this.manager.destroy();
+        if (this.manager) {
+            this.manager.destroy();
+        }
         this.touchMoveSubscription.unsubscribe();
     }
     ngAfterViewInit() {
diff --git a/node_modules/ngx-joystick/fesm2022/ngx-joystick.mjs b/node_modules/ngx-joystick/fesm2022/ngx-joystick.mjs
index d7ef592..2dcbf07 100644
--- a/node_modules/ngx-joystick/fesm2022/ngx-joystick.mjs
+++ b/node_modules/ngx-joystick/fesm2022/ngx-joystick.mjs
@@ -40,7 +40,9 @@ class NgxJoystickComponent {
     }
     ngOnDestroy() {
         window.clearInterval(this.interval);
-        this.manager.destroy();
+        if (this.manager) {
+            this.manager.destroy();
+        }
         this.touchMoveSubscription.unsubscribe();
     }
     ngAfterViewInit() {
```

<em>This issue body was [partially generated by patch-package](https://github.com/ds300/patch-package/issues/296).</em>

**Comment by wweitzel** [comment]
Hi! Thanks for reporting! We just published a new version, 4.0.3, that should fix this. Please try the newest version

**Comment by wweitzel** [comment]
Closing since this should be fixed
